unidist/core/backends/mpi/core/shared_store.py

```diff
@@ -0,0 +1,753 @@
+        # Service buffer includes service information about written shared data.
+        # The service info is set by the worker who sends the data to shared memory
+        # and is removed by the monitor if the data is cleared.
+        # The service info indicates that the current data is written to shared memory
+        # and shows the actual location and number of references.
+        self.service_buffer = None
+        # Length of shared memory buffer
+        self.shared_memory_size = None
+        # Length of service info in service buffer
+        self.service_info_max_count = None
+
+        # Initialize all properties above
+        if common.is_used_shared_memory():
+            self._init_shared_memory()
+
+        # Logger will be initialized after `communicator.MPIState`
+        self.logger = None
+        # Shared memory range {DataID: dict}
+        # Shared information is the necessary information to properly deserialize data from shared memory.
+        self._shared_info = weakref.WeakKeyDictionary()
+        # The list of `weakref.finalize` which should be canceled before closing the shared memory.
+        self.finalizers = []
+
+    def _get_allowed_memory_size(self):
+        """
+        Get allowed memory size for allocate shared memory.
+
+        Returns
+        -------
+        int
+            The number of bytes available to allocate shared memory.
+        """
+        virtual_memory = psutil.virtual_memory().total
+        if sys.platform.startswith("linux"):
+            shm_fd = os.open("/dev/shm", os.O_RDONLY)
+            try:
+                shm_stats = os.fstatvfs(shm_fd)
+                system_memory = shm_stats.f_bsize * shm_stats.f_bavail
+                if system_memory / (virtual_memory / 2) < 0.99:
+                    print(
+                        f"The size of /dev/shm is too small ({system_memory} bytes). The required size "
+                        + f"at least half of RAM ({virtual_memory // 2} bytes). Please, delete files in /dev/shm or "
+                        + "increase size of /dev/shm with --shm-size in Docker."
+                    )
+            finally:
+                os.close(shm_fd)
+        else:
+            system_memory = virtual_memory
+        return system_memory
+
+    def _init_shared_memory(self):
+        """
+        Shared memory initializing
+        """
+        mpi_state = communication.MPIState.get_instance()
+
+        # use only 95% of available memory because the rest is needed for local storages of workers
+        # Shared memory is allocated only once by the monitor process
+        self.shared_memory_size = (
+            int(self._get_allowed_memory_size() * 0.95)
+            if mpi_state.is_monitor_process()
+            else 0
+        )
+
+        info = MPI.Info.Create()
+        info.Set("alloc_shared_noncontig", "true")
+        self.win = MPI.Win.Allocate_shared(
+            self.shared_memory_size * MPI.BYTE.size,
```

Comment:
```suggestion
            self.shared_memory_size,
```
We don't to multiply this because shared_memory_size is already in bytes.

Reply:
Yes, it does not matter, but I added it on purpose that we will not forgot about it if this type is changed.

Reply:
If the data size changes, we will not be able to allocate memory because `shared_memory_size` is already in bytes and we take it in total.

Reply:
In this case we can divide by the itemsize, but in my opinion the current option is more clear.